eve/endpoints.py

```diff
@@ -174,28 +176,76 @@ def media_endpoint(_id):
 
     .. versionadded:: 0.6
     """
-    file_ = app.media.get(_id)
-    if file_ is None:
-        return abort(404)
-
-    if_modified_since = weak_date(request.headers.get('If-Modified-Since'))
-    if if_modified_since is not None:
-        if if_modified_since.tzinfo is None:
-            if_modified_since = if_modified_since.replace(
-                tzinfo=tz_util.utc)
-
-        if if_modified_since > file_.upload_date:
-            return Response(status=304)
-
-    headers = {
-        'Last-Modified': date_to_rfc1123(file_.upload_date),
-        'Content-Length': file_.length,
-    }
-
-    response = Response(file_, headers=headers, mimetype=file_.content_type,
-                        direct_passthrough=True)
-
-    return response
+    range_header = request.headers.get('Range', None)
+    if not range_header:
+        file_ = app.media.get(_id)
+        if file_ is None:
+            return abort(404)
+
+        if_modified_since = weak_date(request.headers.get('If-Modified-Since'))
+        if if_modified_since is not None:
+            if if_modified_since.tzinfo is None:
+                if_modified_since = if_modified_since.replace(
+                    tzinfo=tz_util.utc)
+
+            if if_modified_since > file_.upload_date:
+                return Response(status=304)
+
+        headers = {
+            'Last-Modified': date_to_rfc1123(file_.upload_date),
+            'Content-Length': file_.length,
+            'Accept-Ranges': 'bytes',
+        }
+
+        response = Response(
+            file_,
+            status=200,
+            headers=headers,
+            mimetype=file_.content_type,
+            direct_passthrough=True
+        )
+
+        return response
+    else:
+        file_ = app.media.get(_id)
+        size = file_.length
+        byte1, byte2 = 0, None
+
+        m = re.search('(\d+)-(\d*)', range_header)
+        g = m.groups()
+
+        if g[0]:
+            byte1 = int(g[0])
+        if g[1]:
+            byte2 = int(g[1])
```

Comment:
`g[1]` might be empty string, which will break `int('')`:
```
>>> int('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid literal for int() with base 10: ''
```

Reply:
@nicolaiarocci Yes, I will update the test as soon as possible.

@peterdemin Thank you for correcting me.